File: WOL_ASPDotNet/Utilities/Implement/NetAddressUtility.cs
using System.Text.RegularExpressions;

namespace WOL_ASPDotNet.Utilities.Implement
{
    public static class NetAddressUtility
    {
        public static string FormatMacAddress(string str)
        {
            string result = "";
            char spliter = ':';
            string normalizedStr = Regex.Replace(str, "[^\\w]+", "");

            if (normalizedStr.Length == 12)
            {
                int itrTimes = 12 / 2;

                for (int i = 0; i < itrTimes; i++)
                {
                    result += normalizedStr.Substring(i * 2, 2) + spliter;
                }

                result = result.TrimEnd(spliter);
            }
            else
            {
                throw new ArgumentException("Invalid mac address");
            }

            return result;
        }
    }
}

File: WOL_ASPDotNet/ClientSrc/utilities/NetAddressUtility.ts
import * as lodash from 'lodash';

export const formatMacAddress = (str: string) => {

    let result: string = "";
    const spliter = ":";
    const normalizedStr = lodash.replace(str, /[^\w]+/gm, "");


    if (normalizedStr.length == 12) {
        const itrTimes = 12 / 2;

        for (let i = 0; i < itrTimes; i++) {
            const sIdx: number = i * 2;
            const eIdx: number = i * 2 + 2;
            result += normalizedStr.substring(sIdx, eIdx) + spliter;
        }

        result = lodash.trimEnd(result, spliter);    
    } else {
        throw new Error("Not a valid mac address");
    }

    return result;
}